### PR TITLE
[libosip2] Update to 5.2.0 and add windows support

### LINKED ISSUE #17309: libosip2 build failure
**Host Environment**
- OS:  Windows 
- Compiler: Latest MSVC 2019

**To Reproduce**
`.\vcpkg.exe install libosip2  --triplet x64-windows`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
```
Building package libosip2[core]:x64-windows...
-- Using cached D:/Backup/Code/Qt/krita-vcpkg/downloads/libosip2-5.1.0.tar.gz
-- Cleaning sources at D:/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/src/libosip2-5-69334afe95.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source D:/Backup/Code/Qt/krita-vcpkg/downloads/libosip2-5.1.0.tar.gz
-- Using source at D:/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/src/libosip2-5-69334afe95.clean
-- Configuring x64-windows
-- Using msys root at D:/Backup/Code/Qt/krita-vcpkg/downloads/tools/msys2/d589ebc2087b2b2c
-- Using msys root at D:/Backup/Code/Qt/krita-vcpkg/downloads/tools/msys2/50864d9fae363d6e
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Using msys root at D:/Backup/Code/Qt/krita-vcpkg/downloads/tools/msys2/969c0913b9df89e1
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:146 (message):
    Command failed: D:/Backup/Code/Qt/krita-vcpkg/downloads/tools/msys2/d589ebc2087b2b2c/usr/bin/make.exe -j 33 --trace -f Makefile all
    Working Directory: D:/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/x64-windows-dbg
    See logs for more information:
      D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\build-x64-windows-dbg-out.log
      D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\build-x64-windows-dbg-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_make.cmake:195 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_make.cmake:26 (vcpkg_build_make)
  ports/libosip2/portfile.cmake:22 (vcpkg_install_make)
  scripts/ports.cmake:142 (include)
```
-Please attach any additional failure logs mentioned in the console output.

```
D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\src\libosip2-5-69334afe95.clean\src\osipparser2\sdp_message.c(332) : note: feeds call on this line
D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\src\libosip2-5-69334afe95.clean\src\osipparser2\sdp_message.c(1309) : warning C5045: Compiler will insert Spectre mitigation for memory load if /Qspectre switch specified
D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\src\libosip2-5-69334afe95.clean\src\osipparser2\sdp_message.c(1308) : note: index 'slash' range checked by comparison on this line
D:\Backup\Code\Qt\krita-vcpkg\buildtrees\libosip2\src\libosip2-5-69334afe95.clean\src\osipparser2\sdp_message.c(1309) : note: feeds call on this line
Makefile:461: update target 'libosipparser2.la' due to: osip_proxy_authorization.lo osip_cseq.lo osip_record_route.lo osip_route.lo osip_to.lo osip_from.lo osip_uri.lo osip_authorization.lo osip_header.lo osip_www_authenticate.lo osip_via.lo osip_body.lo osip_md5c.lo osip_message.lo osip_list.lo osip_call_id.lo osip_message_parse.lo osip_contact.lo osip_message_to_str.lo osip_content_length.lo osip_parser_cfg.lo osip_content_type.lo osip_proxy_authenticate.lo osip_mime_version.lo osip_port.lo osip_call_info.lo osip_content_disposition.lo osip_accept_encoding.lo osip_content_encoding.lo osip_authentication_info.lo osip_proxy_authentication_info.lo osip_accept_language.lo osip_accept.lo osip_alert_info.lo osip_error_info.lo osip_allow.lo sdp_accessor.lo sdp_message.lo
/bin/sh ../../libtool  --tag=CC   --mode=link /cygdrive/d/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/src/libosip2-5-69334afe95.clean/scripts/compile cl.exe   -DENABLE_TRACE -nologo -W3 -utf-8 -MP -MDd -Z7 -Ob0 -Od -Xcompiler -RTC1 -version-info 12:0:0   -no-undefined -LD:/Backup/Code/Qt/krita-vcpkg/installed/x64-windows/debug/lib -LD:/Backup/Code/Qt/krita-vcpkg/installed/x64-windows/debug/lib/manual-link -o libosipparser2.la -rpath /D/Backup/Code/Qt/krita-vcpkg/installed/x64-windows/debug/lib osip_proxy_authorization.lo osip_cseq.lo osip_record_route.lo osip_route.lo osip_to.lo osip_from.lo osip_uri.lo osip_authorization.lo osip_header.lo osip_www_authenticate.lo osip_via.lo osip_body.lo osip_md5c.lo osip_message.lo osip_list.lo osip_call_id.lo osip_message_parse.lo osip_contact.lo osip_message_to_str.lo osip_content_length.lo osip_parser_cfg.lo osip_content_type.lo osip_proxy_authenticate.lo osip_mime_version.lo osip_port.lo osip_call_info.lo osip_content_disposition.lo osip_accept_encoding.lo osip_content_encoding.lo osip_authentication_info.lo osip_proxy_authentication_info.lo osip_accept_language.lo osip_accept.lo osip_alert_info.lo osip_error_info.lo osip_allow.lo sdp_accessor.lo sdp_message.lo  -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -lcomdlg32 -ladvapi32
libtool: link: lib.exe cru .libs/libosipparser2.a  osip_proxy_authorization.obj osip_cseq.obj osip_record_route.obj osip_route.obj osip_to.obj osip_from.obj osip_uri.obj osip_authorization.obj osip_header.obj osip_www_authenticate.obj osip_via.obj osip_body.obj osip_md5c.obj osip_message.obj osip_list.obj osip_call_id.obj osip_message_parse.obj osip_contact.obj osip_message_to_str.obj osip_content_length.obj osip_parser_cfg.obj osip_content_type.obj osip_proxy_authenticate.obj osip_mime_version.obj osip_port.obj osip_call_info.obj osip_content_disposition.obj osip_accept_encoding.obj osip_content_encoding.obj osip_authentication_info.obj osip_proxy_authentication_info.obj osip_accept_language.obj osip_accept.obj osip_alert_info.obj osip_error_info.obj osip_allow.obj sdp_accessor.obj sdp_message.obj
Microsoft (R) Library Manager Version 14.28.29913.0
Copyright (C) Microsoft Corporation.  All rights reserved.

LINK : fatal error LNK1181: cannot open input file 'cru'
make[3]: Leaving directory '/cygdrive/d/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/x64-windows-dbg/src/osipparser2'
make[2]: Leaving directory '/cygdrive/d/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/x64-windows-dbg/src'
make[1]: Leaving directory '/cygdrive/d/Backup/Code/Qt/krita-vcpkg/buildtrees/libosip2/x64-windows-dbg'

```

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by NancyLi1013**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes #17309

Update to latest version 5.2.0 and also add support for Windows platform.

`arm `and `arm64 `are not supported on Windows platform.

Note: No feature needs to test.

### CODE REVIEW

**Review by JackBoosY — APPROVED** [review]

**Comment by JackBoosY** [comment]
LGTM.

File: ports/libosip2/portfile.cmake [hunk 1]
```diff
@@ -1,28 +1,56 @@
...
-if(VCPKG_TARGET_IS_WINDOWS)
-    set(OPTIONS --enable-mt=no)
+if(VCPKG_TARGET_IS_WINDOWS)   
+    if (TRIPLET_SYSTEM_ARCH MATCHES "x86")
+        set(BUILD_ARCH "Win32")
+    elseif(TRIPLET_SYSTEM_ARCH MATCHES "x64")
+        set(BUILD_ARCH "x64")
+    elseif(TRIPLET_SYSTEM_ARCH MATCHES "arm")
+        message(FATAL_ERROR " ARM is currently not supported.")
+    endif()
```
Comment (strega-nil-ms) [inline ports/libosip2/portfile.cmake:26]:
```suggestion
    vcpkg_fail_port_install(ON_ARCH "arm" "arm64")
```

File: ports/libosip2/portfile.cmake [hunk 2]
```diff
@@ -1,28 +1,56 @@
...
+        set(BUILD_ARCH "Win32")
+    elseif(TRIPLET_SYSTEM_ARCH MATCHES "x64")
+        set(BUILD_ARCH "x64")
+    elseif(TRIPLET_SYSTEM_ARCH MATCHES "arm")
+        message(FATAL_ERROR " ARM is currently not supported.")
+    endif()    
+    vcpkg_install_msbuild(
+        SOURCE_PATH "${SOURCE_PATH}"
+        PROJECT_SUBPATH "platform/vsnet/osip2.vcxproj"
+        PLATFORM ${BUILD_ARCH}
```
Comment (strega-nil-ms) [inline ports/libosip2/portfile.cmake:30]:
```suggestion
```

`vcpkg_install_msbuild` already does this correctly for this port.

File: ports/libosip2/portfile.cmake [hunk 3]
```diff
@@ -1,28 +1,56 @@
...
+        INCLUDES_SUBPATH include
+        USE_VCPKG_INTEGRATION
+        ALLOW_ROOT_INCLUDES
+        REMOVE_ROOT_INCLUDES      
+    )    
+    vcpkg_install_msbuild(
+        SOURCE_PATH "${SOURCE_PATH}"
+        PLATFORM ${BUILD_ARCH}
```
Comment (strega-nil-ms) [inline ports/libosip2/portfile.cmake:38]:
```suggestion
```

File: ports/libosip2/portfile.cmake [hunk 4]
```diff
@@ -1,28 +1,56 @@
...
+    elseif(TRIPLET_SYSTEM_ARCH MATCHES "arm")
+        message(FATAL_ERROR " ARM is currently not supported.")
+    endif()    
+    vcpkg_install_msbuild(
+        SOURCE_PATH "${SOURCE_PATH}"
+        PROJECT_SUBPATH "platform/vsnet/osip2.vcxproj"
+        PLATFORM ${BUILD_ARCH}
+        INCLUDES_SUBPATH include
+        USE_VCPKG_INTEGRATION
+        ALLOW_ROOT_INCLUDES
```
Comment (strega-nil-ms) [inline ports/libosip2/portfile.cmake:33]:
```suggestion
```

not necessary, since `REMOVE_ROOT_INCLUDES` is passed.

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
Some changes are required to the calls to `vcpkg_install_msbuild`

**Comment by NancyLi1013** [comment]
Thanks for your review suggestions @strega-nil-ms 

All have been done!

File: ports/libosip2/portfile.cmake [hunk 5]
```diff
@@ -1,28 +1,52 @@
...
+        REMOVE_ROOT_INCLUDES      
+    )
+    
+    vcpkg_install_msbuild(
+        SOURCE_PATH "${SOURCE_PATH}"
+        PROJECT_SUBPATH "platform/vsnet/osipparser2.vcxproj"
+        USE_VCPKG_INTEGRATION
+    )
+
+elseif(VCPKG_TARGET_IS_LINUX OR VCPKG_TAREGT_IS_OSX)
```
Comment (NancyLi1013) [inline ports/libosip2/portfile.cmake:37]:
```suggestion
elseif(VCPKG_TARGET_IS_LINUX OR VCPKG_TARGET_IS_OSX)
```

**Comment by strega-nil-ms** [comment]
Thanks @NancyLi1013 :D

**Pull request merged**